### DISCUSSION TITLE
Share your Suggestions, Feedback & Improvements for DB here.

### DISCUSSION THREAD

**Discussion by dr5hn** [dr5hn/countries-states-cities-database#152] [Ideas]
Hello All 👋🏼 ,

I request everyone to share your feedback, suggestions & improvement for this DB. You can also shoot your ideas about what else you want in this DB ?

**Comment by abdulin** [comment]


**Reply by dr5hn** [reply]
@abdulin Thanks for your findings. 
Can you tell me the name of the state which is missing these cities?

**Comment by ionecum** [comment]
Hello, it would be great to add the population data, especially for cities and states. This would allow to sort cities by population, for example, so, in a drop down list, the most important or populated cities would come first. And this would be great in many applications. Notice that even if you have a capital field in the country table, this would not be enough. Washington DC is the capital of the United States but New Youk or Los Angeles are much more important. Therefore a population fields has a higher weight than mentioning a single capital. 

For the countries, it doesn't make sense to sort countries by population, but we could add an "important_flag" field in the database so to allow to add a flag for the most significant countries for a given business. 

Finally it is strongly suggested to name, at least the western cities with their original name. For example "Ciudad de México" instead of "Mexico City".

**Comment by cihanbagdatli** [comment]
#747 We would be very happy if we edit the information here.

**Comment by abbey007** [comment]
Hi i would like to suggest a correction, 

Badajoz  is a city comes under state Extremadura in country Spain, 
But in the city database all the cities which comes under Extremadura have the state Badajoz which is incorrect i think. 

some example cities are 

"Fuente de Cantos"
"Fuente del Arco"
"Ahillones"
"Segura de León"
"Calzadilla de los Barros"
"Aceuchal"
"Fuentes de León"

**Comment by alvarohc777** [comment]
Hi, I would like to know whether you have considered making data available in parquet and/or avro format.

**Comment by abdulrahmanhussin1** [comment]
Hey, nice job on the work! It's been a big help, but I reckon there are a couple of tweaks needed to make it just right for me:

We should split timezone and translation into separate tables.
Let's move the currency stuff into its own table and add columns for exchange rates to handle business needs.

**Comment by ppfoong** [comment]
Ah... I just found this thread.

Copy over my suggestion in the API thread to here.

I suggest to add in a new boolean column in cities table called is_capital and use it to tag whether the city is a capital of its country.

After that, you can update the field in cities table by using this SQL command:

UPDATE cities
SET is_capital = 1
WHERE EXISTS (SELECT * FROM countries WHERE countries.capital = cities.name AND countries.id = cities.country_id);

and then, you can also add a new capital_id column in countries table, and store the capital city's id in that field.

**Comment by pzh20** [comment]
The United Kindom 'States' are in fact local Council Districts within our Counties (States eqv). The UK is devided into Counties as follows;

Bedfordshire
Berkshire
Bristol
Buckinghamshire
Cambridgeshire
Cheshire
Cornwall
County Durham
Cumbria
Derbyshire
Devon
Dorset
East Riding of Yorkshire
East Sussex
Essex
Gloucestershire
Greater London
Greater Manchester
Hampshire
Herefordshire
Hertfordshire
Isle of Wight
Kent
Lancashire
Leicestershire
Lincolnshire
Merseyside
Norfolk
North Somerset
North Yorkshire
Northamptonshire
Northumberland
Nottinghamshire
Oxfordshire
Rutland
Shropshire
Somerset
South Gloucestershire
South Yorkshire
Staffordshire
Suffolk
Surrey
Tyne & Wear
Warwickshire
West Midlands
West Sussex
West Yorkshire
Wiltshire
Worcestershire

Counties in Scotland 2025
Aberdeenshire
Angus
Argyll & Bute
Ayrshire
Banffshire
Berwickshire
Caithness
Clackmannanshire
Dumfries & Galloway
East Ayrshire
East Dunbartonshire
East Lothian
East Renfrewshire
Fife
Highland
Inverclyde
Kincardineshire
Lanarkshire
Midlothian
Moray
North Ayrshire
North Lanarkshire
Orkney
Perth & Kinross
Renfrewshire
Scottish Borders
Shetland
South Ayrshire
South Lanarkshire
Stirlingshire
West Dunbartonshire
West Lothian
Western Isles

List of Counties in Wales 2025
Blaenau Gwent
Bridgend
Caerphilly
Cardiff
Carmarthenshire
Ceredigion
Conwy
Denbighshire
Flintshire
Gwynedd
Isle of Anglesey
Merthyr Tydfil
Monmouthshire
Neath Port Talbot
Newport
Pembrokeshire
Powys
Rhondda Cynon Taff
Swansea
Torfaen
Vale of Glamorgan
Wrexham

List of Northern Ireland Counties 2025
Antrim
Armagh
County Down
Fermanagh
Londonderry
Tyrone

**Comment by yeahfine** [comment]
What I must tell you is that there is a serious error in your administrative division: Taiwan is a part of China and not an independent country!!! Similarly, the Hong Kong Special Administrative Region and the Macau Special Administrative Region are also part of China, and you cannot directly classify them as countries!!! Please correct this error immediately!!!
